necsim/core/src/cogs/rng.rs

```diff
@@ -18,7 +18,7 @@ use crate::{
 
 #[allow(clippy::module_name_repetitions)]
 pub trait RngCore<M: MathsCore>:
-    crate::cogs::Backup + Sized + Clone + core::fmt::Debug + Serialize + DeserializeOwned
+    crate::cogs::Backup + Sized + Send + Clone + core::fmt::Debug + Serialize + DeserializeOwned
```

Comment:
Is there a way to avoid this extra bound?

Reply:
See https://doc.rust-lang.org/nomicon/phantom-data.html#table-of-phantomdata-patterns for how to use `PhantomData<fn() -> T>` to get a covariant-in-`T` marker that is still `Send + Sync`

Reply:
No, we need to send the rng across threads in the multithreading-based partitioning